Change device scan default accumulator type (#724)
ROCm/rocPRIM: CHANGELOG.md

Full documentation for rocPRIM is available at [https://rocm.docs.amd.com/projects/rocPRIM/en/latest/](https://rocm.docs.amd.com/projects/rocPRIM/en/latest/).

## rocPRIM 3.6.0 for ROCm 7.0

### Changed

> * The default scan accumulator types for device-level scan algorithms has be changed. This is a breaking change.

Reviewer: ```suggestion
* The default scan accumulator types for device-level scan algorithms have changed. This is a breaking change.
```

Author: Thanks - I've fixed this up in the new patch set.

> The previous default accumulator types could lead to situations in which unexpected overflow occured (for example,

Reviewer: ```suggestion
The previous default accumulator types could lead to situations in which unexpected overflow occured, such as
```

Author: Fixed, thanks.

> if the input or inital type were smaller than the output type).

Reviewer: ```suggestion
when the input or inital type was smaller than the output type.
```

Author: Done.

> The following is the complete list of affected functions and how their default accumulator types are changing:

Reviewer: ```suggestion
This is a complete list of affected functions and how their default accumulator types are changing:
```

Author: Done.

  * `rocprim::inclusive_scan`
    * current default: `class AccType = typename std::iterator_traits<InputIterator>::value_type>`

>     * future default: `class AccType = rocprim::invoke_result_binary_op_t<typename std::iterator_traits<InputIterator>::value_type, BinaryFunction>`

Reviewer: Is this change in 7.0 or upcoming?

Because future implies that this is changing in a release *after* 7.0.

If it's changing *after* 7.0, then this should be in the upcoming changes section.

If this is changing *in* 7.0, then this should be:

```suggestion
    * new default: `class AccType = rocprim::invoke_result_binary_op_t<typename std::iterator_traits<InputIterator>::value_type, BinaryFunction>`
```

Author: Nice catch, thanks. Yes, this change is happening in 7.0. I thought I'd fixed this locally, but I must have undone my changes before I pushed. I've changed the descriptors to "past" and "new" to make it clearer.
